arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,513 @@
+    config->get_value<std::string>("output_cameras_director") == "" )
+  {
+    KWIVER_CONFIG_FAIL("Config needs value output_cameras_director");
+  }
+  else if ( ! ST::FileIsDirectory(
+    config->get_value<kwiver::vital::path_t>("output_cameras_director") ) )
+  {
+    KWIVER_CONFIG_FAIL("output_cameras_director is not a valid directory");
+  }
+
+  if (!config->has_value("output_landmarks_filename") ||
+    config->get_value<std::string>("output_landmarks_filename") == "" )
+  {
+    KWIVER_CONFIG_FAIL("Config needs value output_landmarks_filename");
+  }
+  else if ( ! ST::FileIsDirectory( ST::CollapseFullPath( ST::GetFilenamePath(
+    config->get_value<kwiver::vital::path_t>("output_landmarks_filename") ))))
+  {
+    KWIVER_CONFIG_FAIL("output_landmarks_filename is not in a valid directory");
+  }
+
+#undef KWIVER_CONFIG_FAIL
+
+  return config_valid;
+}
+} // end namespace
+
+class init_cameras_landmarks::priv
+{
+public:
+  priv()
+  : camera_map_ptr(nullptr),
+    landmark_map_ptr(nullptr),
+    feature_track_set_ptr(nullptr),
+    sfm_constraint_ptr(nullptr),
+    config(nullptr),
+    num_frames(0)
+  {}
+
+  camera_map_sptr camera_map_ptr;
+  landmark_map_sptr landmark_map_ptr;
+  feature_track_set_sptr feature_track_set_ptr;
+  sfm_constraints_sptr sfm_constraint_ptr;
+  initialize_cameras_landmarks_sptr algorithm;
+  kwiver::vital::config_block_sptr config;
+  size_t num_frames;
+
+  enum commandline_mode {SUCCESS, HELP, WRITE, FAIL};
+
+  commandline_mode process_command_line(cxxopts::ParseResult& cmd_args)
+  {
+    static std::string opt_config;
+    static std::string opt_out_config;
+
+    if ( cmd_args["help"].as<bool>() )
+    {
+      return HELP;
+    }
+    if ( cmd_args.count("config") )
+    {
+      opt_config = cmd_args["config"].as<std::string>();
+    }
+    if ( cmd_args.count("output-config") > 0 )
+    {
+      opt_out_config = cmd_args["output-config"].as<std::string>();
+    }
+
+    // Set up top level configuration w/ defaults where applicable.
+    config = default_config();
+
+    // If -c/--config given, read in confg file, merge in with default just
+    // generated
+    if( ! opt_config.empty() )
+    {
+      config->merge_config(kwiver::vital::read_config_file(opt_config));
+    }
+
+    bool valid_config = check_config(config);
+
+    if( ! opt_out_config.empty() )
+    {
+      write_config_file(config, opt_out_config );
+      if(valid_config)
+      {
+        LOG_INFO(main_logger,
+                 "Configuration file contained valid parameters and may be "
+                 "used for running");
+      }
+      else
+      {
+        LOG_WARN(main_logger, "Configuration deemed not valid.");
+      }
+      config = nullptr;
+      return WRITE;
+    }
+    else if(!valid_config)
+    {
+      LOG_ERROR(main_logger, "Configuration not valid.");
+      config = nullptr;
+      return FAIL;
+    }
+
+    return SUCCESS;
+  }
+
+  void register_algorithms()
```

Comment:
This is not needed (and has no effect) because all plugins are loaded by the main kwiver program. Applets are loadable, so everything must be loaded before this applet can run.